### PR TITLE
Support prompting for password with SSR

### PR DESCRIPTION
**Pull request by justinabrokwah-okta**
## Description:

- If password with SSR is enabled, the SIW needs to support it
  - A password field is shown on the `enroll-profile` view
    - Field contains an eye icon to allow user to toggle their password input
  - Password requirements are also shown under the password field

### Bacon test
- With okta-core master [Bacon](https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&sha=f08c21716dac8e3d37450413f0d1b727b7ebcbf2) 🟢 

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
[Box folder with screenshots and video](https://okta.box.com/s/3fe5cyignfxl6dj3wdr17a0t9a0kkcdq)

### Issue:

- [Working demo after okta-core changes merged](https://okta.box.com/s/iktieq32f093me6v8gz2r9nl5kyq2oyd)  
- [OKTA-449726](https://oktainc.atlassian.net/browse/OKTA-449726)

### CODE REVIEW

File: test/testcafe/spec/EnrollProfileView_spec.js
Comment (fangshi-okta) [inline test/testcafe/spec/EnrollProfileView_spec.js:187]: Maybe add a test to make sure the form can be submitted successfully? Also was wondering do we want a Re-enter password field to allow end user to confirm their password. Overall looks good!

Comment (justinabrokwah-okta) [reply test/testcafe/spec/EnrollProfileView_spec.js:187]: Thanks for the review. Yes, I agree about the test; I don't think the backend is fully finished yet but I have a ticket here for end to end testing when it is: https://oktainc.atlassian.net/browse/OKTA-480990

Regarding the `confirm password` field, design confirmed having 1 password field in the [Figma](https://www.figma.com/file/R2S1ukzhwQnScwT0wz3ten/Sign-up?node-id=2%3A331)

File: src/v2/view-builder/views/EnrollProfileView.js
Comment (shreyasbhivandkar-okta) [inline src/v2/view-builder/views/EnrollProfileView.js:124]: nit - since the `displayPasswordPolicy(policy)` depends on the `getPasswordPolicySettings()`, you can consider calling this in the `getPasswordPolicySettings()`. You can therefore skip returning the settings on line#137 and use it as a boolean to call the `displayPasswordPolicy()`

Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/EnrollProfileView.js]: Thanks again for the catch! Addressed

File: playground/mocks/config/responseConfig.js
Comment (shreyasbhivandkar-okta) [inline playground/mocks/config/responseConfig.js:108]: nit: is this intended? Can you consider removing the commented out code?

**Review by shreyasbhivandkar-okta — APPROVED** [review]
LGTM 🚀 
Left a small nit comment

File: playground/mocks/config/responseConfig.js
Comment (justinabrokwah-okta) [reply playground/mocks/config/responseConfig.js:108]: Yes, this was intended - to allow for users to visualize what it's like to have the password prompt on the first page of SSR, waiting for a review from the SIW team if we can keep it in

File: packages/@okta/i18n/src/properties/login.properties
Comment (pradeepdewda-okta) [inline packages/@okta/i18n/src/properties/login.properties:1268]: use existing key `oie.password.label`

File: src/v2/view-builder/views/password/EnrollAuthenticatorPasswordView.js
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/password/EnrollAuthenticatorPasswordView.js:1]: All though it's not part of the PR but we should not use a function that is part of v1 widget.. `getPasswordComplexityDescriptionForHtmlList` 

Either bring it in v2 or write a new one.. You can create a jira for this as tech debt and assign to your team

File: src/v2/view-builder/views/password/PasswordPolicyUtil.js
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/password/PasswordPolicyUtil.js:4]: Since we are doing this and it's new to v2.. can we enhance this pretty old style of showing password requirements.. 

As use types in we should update the requirements are being met or not.. by update color of the strings to red or green.

Tech debt can be solved in a new ticket..

Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/password/PasswordPolicyUtil.js:4]: Ticket created: https://oktainc.atlassian.net/browse/OKTA-484798

File: packages/@okta/i18n/src/properties/login.properties
Comment (justinabrokwah-okta) [reply packages/@okta/i18n/src/properties/login.properties]: Addressed

File: src/v2/view-builder/views/password/EnrollAuthenticatorPasswordView.js
Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/password/EnrollAuthenticatorPasswordView.js:1]: Created ticket to track this: https://oktainc.atlassian.net/browse/OKTA-484804

**Review by pradeepdewda-okta — APPROVED** [review]

File: playground/mocks/data/idp/idx/enroll-profile-with-password-returns-error.json
Comment (aarongranick-okta) [inline playground/mocks/data/idp/idx/enroll-profile-with-password-returns-error.json:2]: use a real version here, "1.0.0"

File: playground/mocks/data/idp/idx/enroll-profile-with-password.json
Comment (aarongranick-okta) [inline playground/mocks/data/idp/idx/enroll-profile-with-password.json:3]: same comment as above, client library will validate version, use a real version here, "1.0.0"

Comment (justinabrokwah-okta) [reply playground/mocks/data/idp/idx/enroll-profile-with-password.json]: Thanks, changed in both places indicated

**Review by aarongranick-okta — APPROVED** [review]

**Pull request closed without merging**
